bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/handler/SensiboAccountHandler.java

```diff
@@ -0,0 +1,328 @@
+        super(bridge);
+        this.httpClient = httpClient;
+
+        gson = new GsonBuilder().registerTypeAdapter(ZonedDateTime.class, new TypeAdapter<ZonedDateTime>() {
+            @Override
+            public void write(final @NonNullByDefault({}) JsonWriter out, final ZonedDateTime value)
+                    throws IOException {
+                out.value(value.toString());
+            }
+
+            @Override
+            public ZonedDateTime read(final @NonNullByDefault({}) JsonReader in) throws IOException {
+                return ZonedDateTime.parse(in.nextString());
+            }
+        }).setLenient().setPrettyPrinting().create();
+
+        requestLogger = new RequestLogger(bridge.getUID().getId(), gson);
+    }
+
+    private boolean allowModelUpdate() {
+        final long diffMsSinceLastUpdate = System.currentTimeMillis() - model.getLastUpdated();
+        return diffMsSinceLastUpdate > MIN_TIME_BETWEEEN_MODEL_UPDATES_MS;
+    }
+
+    public SensiboModel getModel() {
+        return model;
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        // Ignore commands as none are supported
+    }
+
+    public SensiboAccountConfiguration loadConfigSafely() throws SensiboConfigurationException {
+        SensiboAccountConfiguration loadedConfig = getConfigAs(SensiboAccountConfiguration.class);
+        if (loadedConfig == null) {
+            throw new SensiboConfigurationException("Could not load Sensibo account configuration");
+        }
+
+        return loadedConfig;
+    }
+
+    @Override
+    public void initialize() {
+        scheduler.execute(() -> {
+            try {
+                config = loadConfigSafely();
+                logger.debug("Initializing Sensibo Account bridge using config {}", config);
+                model = refreshModel();
+                updateStatus(ThingStatus.ONLINE);
+                initPolling();
+                logger.debug("Initialization of Sensibo account completed successfully for {}", config);
+            } catch (final SensiboConfigurationException e) {
+                logger.info("Error initializing Sensibo data: {}", e.getMessage());
+                model = new SensiboModel(0); // Empty model
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Error fetching initial data: " + e.getMessage());
+            } catch (final SensiboException e) {
+                logger.info("Error initializing Sensibo data: {}", e.getMessage());
+                model = new SensiboModel(0); // Empty model
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Error fetching initial data: " + e.getMessage());
+                // Reschedule init
+                scheduler.schedule(this::initialize, 30, TimeUnit.SECONDS);
+            }
```

Comment:
I suggest putting this into its own method so that rescheduling doesn't waste a scheduled task by just scheduling another task.

Reply:
Fixed